contents/docs/sql/variables.mdx

```diff
@@ -0,0 +1,52 @@
+---
+title: SQL variables
+sidebar: Docs
+showTitle: true
+availability:
+  free: full
+  selfServe: full
+  enterprise: full
+---
+
+export const WarehouseLight = "https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2025_02_24_at_14_05_39_2x_5a280bacfd.png"
+export const WarehouseDark = "https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2025_02_24_at_14_05_53_2x_32a95cbe79.png"
+
+
+SQL variables allow you to create configurable, reusable queries by defining placeholders that can be dynamically replaced with different values.
+
+## Using variables in SQL queries
+
+Variables in PostHog SQL are defined using the `{variables.<variable-name>}` syntax. When you write a query with variables, you'll be prompted to provide values for each variable.
+
+For example, this query uses a variable to filter events by a specific date range:
+
+```sql
+SELECT 
+    event,
+    count() as count
+FROM events
+WHERE timestamp >= {variables.startdate}
+    AND timestamp <= {variables.enddate}
+GROUP BY event
+ORDER BY count DESC
+LIMIT 10
+```
```

Comment:
It may be worth adding that you need to create a variable first before being able to use them

Reply:
Also, do we want to set a precedent for using snake case for variable names maybe? `{variables.start_date}`?

Reply:
right now they're getting squashed and the snake case is being removed. So i'm going to merge as is until we investigate/fix that